main.go

```diff
@@ -0,0 +1,168 @@
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	cloudingressmanagedopenshiftiov1alpha1 "github.com/openshift/cloud-ingress-operator/api/v1alpha1"
+	apischemecontroller "github.com/openshift/cloud-ingress-operator/controllers/apischeme"
+	publishingstrategycontroller "github.com/openshift/cloud-ingress-operator/controllers/publishingstrategy"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+var (
+	osdMetricsPort = "8181"
+	osdMetricsPath = "/metrics"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(cloudingressmanagedopenshiftiov1alpha1.AddToScheme(scheme))
+
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(machineapi.AddToScheme(scheme))
+	utilruntime.Must(ingresscontroller.AddToScheme(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+	utilruntime.Must(awsproviderapi.SchemeBuilder.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "f3d6b689.cloudingress.managed.openshift.io",
+		Namespace:              config.OperatorNamespace,
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// setup apischemecontroller with mgr
+	if err = (&apischemecontroller.APISchemeReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "APIScheme")
+		os.Exit(1)
+	}
+
+	// setup publishingstrategycontroller with mgr
+	if err = (&publishingstrategycontroller.PublishingStrategyReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "PublishingStrategy")
+		os.Exit(1)
+	}
+
```

Comment:
Adding the router service controller to the manager seems to have been missed.

Reply:
I think so... the routerservice controller added to the manager now